services/motion/builtin/builtin.go

```diff
@@ -432,3 +392,92 @@ func (ms *builtIn) GetPose(
 		supplementalTransforms,
 	)
 }
+
+// PlanMoveOnMap returns the plan for MoveOnMap to execute
+func (ms *builtIn) planMoveOnMap(
```

Comment:
[q] Should this be split into sub functions? I leave it up to motion to decide

Reply:
@raybjork

Reply:
This is fine.  Eventually we'll want to do a refactor of MoveOnMap and MoveOnGlobe to consolidate but not here